### ISSUE TITLE
Native collection memory leak in AssetHttpClient

### ISSUE THREAD

**Issue by dbirman** [Jirka-Mayer/UnisaveAsset#3]
Thanks for building this code, seems like maybe it isn't being maintained but I thought I would send this issue anyways. This code in `AssetHttpClient` causes a memory leak. Full error at the bottom.

```
UploadHandler uploadHandler = null;
if (payload != null)
    uploadHandler = new UploadHandlerRaw(
        Encoding.UTF8.GetBytes(payload.ToString())
    );
```

I believe this is easily fixed by just calling `uploadHandler.Dispose()` after the request is sent. 

Here's the full stack trace if it's helpful:
```
A Native Collection has not been disposed, resulting in a memory leak. Allocated from:
Unity.Collections.NativeArray`1:.ctor(Byte[], Allocator)
UnityEngine.Networking.UploadHandlerRaw:.ctor(Byte[])
Unisave.HttpClient.<TheRequestCoroutine>d__6:MoveNext() (at Assets\Unisave\Scripts\HttpClient\AssetHttpClient.cs:98)
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
UnityEngine.MonoBehaviour:StartCoroutineManaged2(MonoBehaviour, IEnumerator)
UnityEngine.MonoBehaviour:StartCoroutine(IEnumerator)
Unisave.HttpClient.HttpClientComponent:SendRequest(String, String, Dictionary`2, JsonObject, Action`1) (at Assets\Unisave\Scripts\HttpClient\HttpClientComponent.cs:62)
Unisave.HttpClient.AssetHttpClient:Send(String, String, Dictionary`2, JsonObject, Action`1) (at Assets\Unisave\Scripts\HttpClient\AssetHttpClient.cs:75)
Unisave.HttpClient.AssetHttpClient:Post(String, JsonObject, Action`1) (at Assets\Unisave\Scripts\HttpClient\AssetHttpClient.cs:52)
Unisave.Facets.UnisaveFacetCaller:PerformFacetCall(String, String, JsonArray) (at Assets\Unisave\Scripts\Facets\UnisaveFacetCaller.cs:39)
Unisave.Facets.FacetCaller:CallFacetMethod(Type, Type, String, Object[]) (at Assets\Unisave\Scripts\Facets\FacetCaller.cs:96)
Unisave.Facets.FacetCaller:CallFacetMethod(String, Object[]) (at Assets\Unisave\Scripts\Facets\FacetCaller.cs:51)
Unisave.Facades.OnFacet`1:Call(String, Object[]) (at Assets\Unisave\Scripts\Facades\OnFacet.cs:32)
Unisave.Facades.OnFacet`1:CallAsync(String, Object[]) (at Assets\Unisave\Scripts\Facades\OnFacet.cs:47)
<OnLoginClicked>d__6:MoveNext() (at Assets\Scripts\Accounts\EmailLoginForm.cs:73)
System.Runtime.CompilerServices.AsyncVoidMethodBuilder:Start(<OnLoginClicked>d__6&)
EmailLoginForm:OnLoginClicked()
UnityEngine.Events.InvokableCall:Invoke()
UnityEngine.Events.UnityEvent:Invoke()
UnityEngine.UI.Button:Press() (at Library\PackageCache\com.unity.ugui@1.0.0\Runtime\UI\Core\Button.cs:70)
UnityEngine.UI.Button:OnPointerClick(PointerEventData) (at Library\PackageCache\com.unity.ugui@1.0.0\Runtime\UI\Core\Button.cs:114)
UnityEngine.EventSystems.ExecuteEvents:Execute(IPointerClickHandler, BaseEventData) (at Library\PackageCache\com.unity.ugui@1.0.0\Runtime\EventSystem\ExecuteEvents.cs:57)
UnityEngine.EventSystems.ExecuteEvents:Execute(GameObject, BaseEventData, EventFunction`1) (at Library\PackageCache\com.unity.ugui@1.0.0\Runtime\EventSystem\ExecuteEvents.cs:272)
UnityEngine.EventSystems.StandaloneInputModule:ReleaseMouse(PointerEventData, GameObject) (at Library\PackageCache\com.unity.ugui@1.0.0\Runtime\EventSystem\InputModules\StandaloneInputModule.cs:190)
UnityEngine.EventSystems.StandaloneInputModule:ProcessMousePress(MouseButtonEventData) (at Library\PackageCache\com.unity.ugui@1.0.0\Runtime\EventSystem\InputModules\StandaloneInputModule.cs:640)
UnityEngine.EventSystems.StandaloneInputModule:ProcessMouseEvent(Int32) (at Library\PackageCache\com.unity.ugui@1.0.0\Runtime\EventSystem\InputModules\StandaloneInputModule.cs:546)
UnityEngine.EventSystems.StandaloneInputModule:ProcessMouseEvent() (at Library\PackageCache\com.unity.ugui@1.0.0\Runtime\EventSystem\InputModules\StandaloneInputModule.cs:526)
UnityEngine.EventSystems.StandaloneInputModule:Process() (at Library\PackageCache\com.unity.ugui@1.0.0\Runtime\EventSystem\InputModules\StandaloneInputModule.cs:280)
UnityEngine.EventSystems.EventSystem:Update() (at Library\PackageCache\com.unity.ugui@1.0.0\Runtime\EventSystem\EventSystem.cs:501)

```

**Comment by Jirka-Mayer** [comment]
Thanks for pointing out the issue! I'm about to focus on the asset again and do major refactoring. So this issue came at the right moment.

**Comment by Jirka-Mayer** [comment]
The issue should be resolved by 5f18961a7e2e771ae982018bca72834b6088fd7d. It will be present in the next asset release.
